wfe/web-front-end.go

```diff
@@ -298,9 +297,22 @@ const (
 	malformedJWS = "Unable to read/verify body"
 )
 
+// verifyPOST reads and parses the request body, looks up the Registration
+// corresponding to its JWK, verifies the JWS signature,
+// checks that the resource field is present and correct in the JWS protected
+// header, and returns the JWS payload bytes, the key used to verify, and the
+// corresponding Registration (or error).
+// If regCheck is false, verifyPOST will still try to look up a registration
+// object, and will return it if found. However, if no registration object is
+// found, verifyPOST will attempt to verify the JWS using the key in the JWS
+// headers, and return the key plus a dummy registration if successful. If a
+// caller passes regCheck = false, it should plan on validating the key itself.
 func (wfe *WebFrontEndImpl) verifyPOST(request *http.Request, regCheck bool, resource core.AcmeResource) ([]byte, *jose.JsonWebKey, core.Registration, error) {
```

Comment:
I like the idea of turning this into a handler that runs before all other handlers, although as you say it may be a little tricksy with the two endpoints that don't require a reg (NewRegistration and RevokeCertificate - see RevokeCertificate impl for why).

I filed an issue for that work: https://github.com/letsencrypt/boulder/issues/880. I'd like to avoid the problem where documenting the complexity of an existing function means blocking a change until that function is refactored. Let's land this change, including the documentation and (what I consider) a complexity-neutral code move, then discuss the more in-depth refactor. I also want to understand how that interplays with using `pat` or some other nicer URL handler.